tapdb/universe.go

```diff
@@ -117,15 +117,17 @@ type BaseUniverseStore interface {
 // for a specific proof type.
 //
 // NOTE: This makes an assumption that only specifiers with a group key are
-// valid.
+// valid for the ignore and burn proof types.
 func specifierToIdentifier(spec asset.Specifier,
 	proofType universe.ProofType) (universe.Identifier, error) {
 
 	var id universe.Identifier
 
 	// The specifier must have a group key to be able to be used within the
 	// ignore or burn tree context.
-	if !spec.HasGroupPubKey() {
+	requireGroupKey := proofType == universe.ProofTypeIgnore ||
```

Comment:
This looks like a useful method that we could add on the `universe.ProofType` type directly.

Reply:
It's less about re-usability and more about signifying that there's a specific restriction for the `universe.ProofTypeIgnore` and `universe.ProofTypeBurn` types, in that they can/should only be produced for grouped assets.

So yeah, a method `func (t ProofType) RequireGroupKey() bool {` would make that more clear in my opinion.
But just a suggestion, non-blocking.